modules/javafx.graphics/src/main/java/javafx/scene/layout/BorderConverter.java

```diff
@@ -162,36 +163,123 @@ public Border convert(Map<CssMetaData<? extends Styleable, ?>, Object> converted
+        CornerRadii[] borderRadius = new CornerRadii[strokesCount];
+        Insets[] borderInsets = new Insets[strokesCount];
+
+        for (int i = 0; i < strokesCount; ++i) {
+            BorderStroke stroke = strokes.get(i);
+            borderStrokeFills[i] = new Paint[4];
+            borderStrokeFills[i][0] = stroke.getTopStroke();
+            borderStrokeFills[i][1] = stroke.getRightStroke();
+            borderStrokeFills[i][2] = stroke.getBottomStroke();
+            borderStrokeFills[i][3] = stroke.getLeftStroke();
+
+            borderStrokeStyles[i] = new BorderStrokeStyle[4];
+            borderStrokeStyles[i][0] = stroke.getTopStyle();
+            borderStrokeStyles[i][1] = stroke.getRightStyle();
+            borderStrokeStyles[i][2] = stroke.getBottomStyle();
+            borderStrokeStyles[i][3] = stroke.getLeftStyle();
+
+            BorderWidths widths = stroke.getWidths();
+            borderWidths[i] = new Margins(widths.getTop(), widths.getRight(), widths.getBottom(), widths.getLeft(), false);
+
+            borderRadius[i] = stroke.getRadii();
+            borderInsets[i] = stroke.getInsets();
+        }
+
+        List<BorderImage> borderImages = border.getImages();
+        int imagesCount = borderImages.size();
+        Image[] images = new Image[imagesCount];
+        RepeatStruct[] imageRepeats = new RepeatStruct[imagesCount];
+        BorderImageSlices[] imageSlices = new BorderImageSlices[imagesCount];
+        BorderWidths[] imageWidths = new BorderWidths[imagesCount];
+        Insets[] imageInsets = new Insets[imagesCount];
+
+        for (int i = 0; i < imagesCount; ++i) {
+            BorderImage image = borderImages.get(i);
+            images[i] = image.getImage();
+            imageRepeats[i] = new RepeatStruct(
+                convertToBackgroundRepeat(image.getRepeatX()),
+                convertToBackgroundRepeat(image.getRepeatY()));
+            imageSlices[i] = new BorderImageSlices(image.getSlices(), image.isFilled());
+            imageWidths[i] = image.getWidths();
+            imageInsets[i] = image.getInsets();
+        }
+
+        return Map.of(
+            Border.BORDER_COLOR, borderStrokeFills,
+            Border.BORDER_STYLE, borderStrokeStyles,
+            Border.BORDER_WIDTH, borderWidths,
+            Border.BORDER_RADIUS, borderRadius,
+            Border.BORDER_INSETS, borderInsets,
+            Border.BORDER_IMAGE_SOURCE, images,
+            Border.BORDER_IMAGE_REPEAT, imageRepeats,
+            Border.BORDER_IMAGE_SLICE, imageSlices,
+            Border.BORDER_IMAGE_WIDTH, imageWidths,
+            Border.BORDER_IMAGE_INSETS, imageInsets);
+    }
+
+    private BackgroundRepeat convertToBackgroundRepeat(BorderRepeat repeat) {
+        return switch (repeat) {
+            case REPEAT -> BackgroundRepeat.REPEAT;
+            case ROUND -> BackgroundRepeat.ROUND;
+            case SPACE -> BackgroundRepeat.SPACE;
+            case STRETCH -> BackgroundRepeat.NO_REPEAT;
```

Comment:
what do you think of adding default: case?
this code will not compile if BorderRepeat ever acquires a new value.

Reply:
Yes, I think that's good, as it will alert us to the impact of adding a new enum value. We would probably need to account for the new value here (there's no good default we can choose).

Reply:
Sure, but that would be a runtime errro as compared to a compile-time error, so we run the risk of missing it.